pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -353,23 +353,32 @@ func (s *DiscoveryServer) updateCluster(push *model.PushContext, clusterName str
 }
 
 // SvcUpdate is a callback from service discovery when service info changes.
-func (s *DiscoveryServer) SvcUpdate(cluster, hostname string, ports map[string]uint32, rports map[uint32]string) {
+func (s *DiscoveryServer) SvcUpdate(cluster, hostname string, ports map[string]uint32, _ map[uint32]string) {
 	inboundUpdates.With(prometheus.Labels{"type": "svc"}).Add(1)
 	pc := s.globalPushContext()
-	// In 1.0 Services and configs are only from the 'primary' K8S cluster.
+	pl := model.PortList{}
+	for k, v := range ports {
+		pl = append(pl, &model.Port{
+			Port: int(v),
+			Name: k,
+		})
+	}
 	if cluster == string(serviceregistry.KubernetesRegistry) {
-		pl := model.PortList{}
-		for k, v := range ports {
-			pl = append(pl, &model.Port{
-				Port: int(v),
-				Name: k,
-			})
-		}
 		pc.Mutex.Lock()
-		pc.ServicePort2Name[hostname] = pl
+		pc.ServicePortByHostname[model.Hostname(hostname)] = pl
+		pc.Mutex.Unlock()
+	} else {
+		pc.Mutex.Lock()
+		ports, ok := pc.ServicePortByHostname[model.Hostname(hostname)]
 		pc.Mutex.Unlock()
+		if ok {
+			if !reflect.DeepEqual(ports, pl) {
+				adsLog.Warnf("service %s within cluster %s does not match the one in primary cluster", hostname, cluster)
+			}
+		} else {
+			adsLog.Debugf("service %s within cluster %s occurs before primary cluster, will only take services within primary cluster", hostname, cluster)
+		}
```

Comment:
Isn’t this called for consul, service entries etc ?

Reply:
no, it is only for k8s